app/feature.go

```diff
@@ -0,0 +1,39 @@
+package app
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/rancher/pkg/features"
+)
+
+func addFeatures(cfg Config) error {
+	if cfg.Features == "" {
+		return nil
+	}
+
+	featuresAndValues := strings.Split(cfg.Features, ",")
+
+	if len(featuresAndValues) == 0 {
+		return nil
+	}
+
+	for _, val := range featuresAndValues {
+		feat := strings.Split(val, "=")
+		if len(feat) != 2 {
+			return httperror.NewAPIError(httperror.InvalidBodyContent, "features value must be in \"featureName=boolValue\" format")
+		}
+
+		name := feat[0]
+		b, err := strconv.ParseBool(feat[1])
+		if err != nil {
+			return httperror.NewAPIError(httperror.InvalidBodyContent, "improper input for features: "+feat[1]+" is not a bool")
```

Comment:
Drop the http error part, this is not going back through the API, it will panic a couple levels up.